shared/src/types/ethereum_events.rs

```diff
@@ -5,24 +5,43 @@ use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
 
 use crate::proto::MultiSigned;
 use crate::types::address::Address;
+use crate::types::hash::Hash;
 use crate::types::token::Amount;
 
+/// Each event has a nonce set by the Ethereum smart contract that emitted it.
+#[derive(
+    Debug, PartialEq, Eq, Clone, BorshSerialize, BorshDeserialize, BorshSchema,
+)]
+pub struct EthereumEventNonce(u64);
```

Comment:
Do we need such a long type name? Could we just get away with `Nonce`?. The type `ethereum_events::Nonce` should make the type clear enough.

Reply:
Yeah `Nonce` is better. Especially considering we can explicitly import as `ethereum_events::Nonce` if we need the extra context in the name.